docs/src/proposals/optimistic-transaction-propagation-signal.md

```diff
@@ -0,0 +1,109 @@
+---
+title: Optimistic Transaction Propagation Signal
+---
+
+## Current Retransmit behavior
+
+The retransmission tree currently considers:
+1. epoch staked nodes
+2. tvu peers (filtered by contact info and shred version)
+3. current validator
+
+concatenating (1), (2), and (3)
+deduplicating this list of entries by pubkey favoring entries with contact info
+filtering this list by entries with contact info
+
+This list is then is randomly shuffled by stake weight.
+
+Shreds are then retransmitted to up to FANOUT neighbors and up to FANOUT
+children.
+
+## Deterministic retransmission tree
+
+`weighted_shuffle` will use a deterministic seed when
+`enable_deterministic_seed` has been enabled based on the triple (shred slot,
+shred index, leader pubkey):
+
+```
+if enable_deterministic_seed(self.slot(), root_bank) {
+    hashv(&[
+        &self.slot().to_le_bytes(),
+        &self.index().to_le_bytes(),
+        &leader_pubkey.to_bytes(),
+    ])
+```
+
+First, only epoch staked nodes will be considered regardless of presence of
```

Comment:
This change will need to be coordinated with some of the RPC providers so that their nodes aren't cut off.

Reply:
I don't think this would substantially change the shred distribution to non-staked nodes. It would mostly change the behavior to staked nodes before the next epoch. I'm proposing that we still distribute shreds to other nodes, they just fall outside of the first pass of the calculation which would become deterministic.